auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue

```diff
@@ -338,16 +451,273 @@ export default defineComponent({
+      }
+
+      if (isNameRequest(item)) {
+        const nrState = status(item)
+        if (nrState !== NrDisplayStates.APPROVED) {
+          return false
+        }
+        const nrRequestActionCd = item.nameRequest?.requestActionCd
+        if (nrRequestActionCd === NrRequestActionCodes.NEW_BUSINESS) {
+          return !isModernizedEntity(item)
+        }
+        // temporarily show external icon for amalgamate and continue in
+        if (nrRequestActionCd === NrRequestActionCodes.AMALGAMATE ||
+          nrRequestActionCd === NrRequestActionCodes.MOVE) {
+          return true
+        }
+        return false
+      }
+
+      // check for business
+      return !isModernizedEntity(item)
+    }
+
+    const isBusinessAffiliated = (businessIdentifier: string): boolean => {
+      return affiliations.results.some(business => businessIdentifier === business.businessIdentifier)
+    }
+
+    const goToAmalgamate = (item: Business): void => {
+      // For now, go to COLIN with external link + icon + matching hover text
+      goToCorpOnline()
+    }
+
+    // Continue In
+    const goToRelocate = (item: Business): void => {
+      goToCorpOnline()
+    }
+
+    const goToFormPage = (): void => {
+      window.location.href = ConfigHelper.getCorpFormsUrl()
+    }
+
+    const goToRegister = async (item: Business): Promise<void> => {
+      if (isModernizedEntity(item)) {
+        const businessIdentifier = await createBusinessRecord(item)
+        goToDashboard(businessIdentifier)
+      } else if (isColinEntity(item)) {
+        goToCorpOnline()
+      } else if (isSocieties(item)) {
+        goToSocieties()
+      } else {
+        goToFormPage()
+      }
+    }
+
+    const redirect = (item: Business): void => {
+      if (isTemporaryBusiness(item)) {
+        goToDashboard(item.businessIdentifier)
+      } else if (isNameRequest(item)) {
+        if (status(item) === NrDisplayStates.APPROVED) {
+          const nrRequestActionCd = item.nameRequest?.requestActionCd
+          switch (nrRequestActionCd) {
+            case NrRequestActionCodes.AMALGAMATE:
+              goToAmalgamate(item)
+              break
+            case NrRequestActionCodes.MOVE:
+              goToRelocate(item)
+              break
+            case NrRequestActionCodes.CONVERSION:
+            case NrRequestActionCodes.CHANGE_NAME:
+            case NrRequestActionCodes.RESTORE:
+            case NrRequestActionCodes.RENEW: {
+              if (isBusinessAffiliated(item.nameRequest?.corpNum)) {
+                goToDashboard(item.nameRequest?.corpNum)
+              } else {
+                let action = ''
+                if (nrRequestActionCd === NrRequestActionCodes.CONVERSION) {
+                  action = 'alter'
+                } else if (nrRequestActionCd === NrRequestActionCodes.CHANGE_NAME) {
+                  action = 'change name'
+                } else {
+                  const action = isAllowRestore(item) ? 'restore' : 'reinstate'
```

Comment:
This declaration will override the variable in the parent scope, but this assignment probably won't be visible in line 644.

Reply:
This is the cause of one code smell (in sonarcloud). I resolved the other one as "won't fix".